src/Resolvers/Microsoft.NET.Sdk.WorkloadResolver/MSBuildWorkloadSdkResolver.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+
+#if NET
+using Microsoft.DotNet.Cli;
+#else
+using Microsoft.DotNet.DotNetSdkResolver;
+#endif
+
+#nullable disable
+
+namespace Microsoft.NET.Sdk.WorkloadResolver
+{
+    public class MSBuildWorkloadSdkResolver : SdkResolver
+    {
+        public override string Name => "Microsoft.DotNet.MSBuildWorkloadSdkResolver";
+
+        public override int Priority => 4000;
+
+        private IWorkloadManifestProvider _workloadManifestProvider;
+        private IWorkloadResolver _workloadResolver;
+
+
+#if NETFRAMEWORK
+        private readonly NETCoreSdkResolver _sdkResolver;
+#endif
+
+        public MSBuildWorkloadSdkResolver()
+        {
+#if NETFRAMEWORK
+            _sdkResolver = new NETCoreSdkResolver();
+#endif
+        }
+
+        private void InitializeWorkloadResolver(SdkResolverContext context)
+        {
+            var dotnetRootPath = GetDotNetRoot(context);
+
+            var sdkDirectory = GetSdkDirectory(context);
+            //  TODO: Is the directory name OK, or should we read the .version file (we don't want to use the Cli.Utils library to read it on .NET Framework)
+            var sdkVersion = Path.GetFileName(sdkDirectory);
```

Comment:
read the folder name is enough. It is a contract used by at least template engine already